sql/moz-fx-data-shared-prod/telemetry_derived/urlbar_clients_daily_v1/query.sql

```diff
@@ -2,12 +2,16 @@ CREATE TEMP FUNCTION one_index(x ANY TYPE) AS (
   CAST(IF(SAFE_CAST(x AS INT64) < 0, SAFE_CAST(x AS INT64), SAFE_CAST(x AS INT64) + 1) AS STRING)
 );
 
-CREATE TEMP FUNCTION one_index_struct(x STRUCT<k STRING, v INT64>) AS (
-  STRUCT(one_index(x.k) AS key, x.v AS value)
+CREATE TEMP FUNCTION one_index_struct(record STRUCT<k STRING, v INT64>) AS (
+  STRUCT(one_index(record.k) AS key, record.v AS value)
 );
 
-CREATE TEMP FUNCTION one_index_array(x ARRAY<STRUCT<k STRING, v INT64>>) AS (
-  ARRAY(SELECT one_index_struct(e) FROM UNNEST(x) AS e)
+CREATE TEMP FUNCTION one_index_array(map ARRAY<STRUCT<k STRING, v INT64>>) AS (
+  ARRAY(SELECT one_index_struct(record) FROM UNNEST(map) AS record)
+);
+
+CREATE TEMP FUNCTION replace_outlier_values_with_zero(map ARRAY<STRUCT<k STRING, v INT64>>) AS (
+    ARRAY(SELECT STRUCT(record.k AS key, IF(record.v > 1000000, 0, record.v)) FROM UNNEST(map) AS record)
```

Comment:
I'd add a comment about this number and why we do this filtering.

Reply:
@rebecca-burwei, do you have any insights to what we should set as a threshold? I was told that the search counts have a similar procedure that's at 20k iirc

Reply:
Here are the limits for [Desktop](https://github.com/mozilla/bigquery-etl/blob/main/sql/moz-fx-data-shared-prod/search_derived/search_clients_daily_v8/query.sql#L263) and [Mobile](https://github.com/mozilla/bigquery-etl/blob/main/sql/moz-fx-data-shared-prod/search_derived/mobile_search_clients_daily_v1/query.sql#L632-L633) search. Looking backwards seems like it was just more than expected for the sum of probe counts from a single client in a single day. 

Looking at a window around the anomaly day by client it was definitely one client with 1B+ probe count sums that day. The second highest was on the order of ~100,000 and a few with ~10,000.

Reply:
ok, cool. i think let's go with 10,000 similar to the above and make a comment about it.

Reply:
just thought about this: should we be mapping this to zero or to NULL? they have very different implications. other patterns mentioned above map it to NULL

Reply:
@teonbrooks what implications are you thinking of?

I do agree however that `NULL` makes sense as opposed to `0`, which signifies "no interactions".

Reply:
mostly when it comes to computing picked per user, if a user actually didn't choose anything, that is worth noting as 0 whereas if we have the behavior of >10k, we would want to remove the user from consideration in the average

Reply:
Agreed, just added a test that revealed that. We could filter the probes that fall outside the threshold altogether?